tools/scripts/ffi.toml

```diff
@@ -312,4 +313,137 @@ else
     echo "Installing Diplomat rev ${rev}"
     exec cargo install --git https://github.com/rust-diplomat/diplomat.git --rev ${rev} diplomat-tool -f
 end
-'''
\ No newline at end of file
+'''
+
+[tasks.gn-install]
+description = "Install tools required for GN locally in the icu4x project"
+category = "ICU4X Development"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+mkdir ffi/gn/third_party_tools
```

Comment:
Good thoughts.

A single top-level directory could work, but I kind-of like making it GN-specific. I don't expect that these tools will be needed more generally. By making it GN-specific, these tools can be installed and updated independently of other potential build tools that other corners of ICU4X might need. Also, I can point the GHA cache path at it.

I'm not a big fan of home-directory paths, because (1) I might not have permissions to write there, (2) I'm worried about blowing things away that aren't mine, and (3) a bit trickier to make it work cross-platform.